crates/typst-library/src/text/misc.rs

```diff
@@ -301,6 +317,46 @@ pub fn smallcaps(
 pub fn lorem(
     /// The length of the blind text in words.
     words: usize,
-) -> Str {
-    lipsum::lipsum(words).replace("--", "–").into()
+    /// Whether to split the blind text into paragraphs or not. Defaults to `false`.
+    #[named]
+    as_paragraphs: Option<bool>,
+) -> StrOrContent {
+    let text = lipsum::lipsum(words).replace("--", "–");
+    if as_paragraphs == Some(true) {
+        // Split into natural paragraphs
+        const SENTENCES_PER_PARAGRAPH: usize = 4;
+
+        let mut elems = Vec::new();
+        let mut last_i = 0;
+        let mut sentence_counter = 0;
+        while last_i < text.len() {
+            let mut i = last_i;
+            loop {
```

Comment:
Does this really need two nested loops? Couldn't the condition of the outer loop be integrated into the inner one?